cli/popper/runner.py

```diff
@@ -32,28 +33,33 @@ def __init__(self, config_file=None, workspace_dir=os.getcwd(),
         self.repo = scm.new_repo()
         self.config.workspace_sha = scm.get_sha(self.repo)
 
+        self.config.engine_name = engine
+        self.config.resman_name = resource_manager
+        self.popper_cfg = DotMap()
+
         if config_file:
-            # read options from config file
-            config_from_file = pu.load_config_file(config_file)
+            self.popper_cfg = PopperConfig(config_file)
 
-            if hasattr(config_from_file, 'ENGINE'):
-                self.config.engine_options = config_from_file.ENGINE
-            if hasattr(config_from_file, 'RESOURCE_MANAGER'):
-                self.config.resman_options = config_from_file.RESOURCE_MANAGER
+        self.config.engine_name = pu.select_not_none(
+            [self.config.engine_name, self.popper_cfg.engine.name, 'docker'])
+        self.config.resman_name = pu.select_not_none(
+            [self.config.resman_name,
+             self.popper_cfg.resource_manager.name,
+             'host'])
 
-        if not self.config.resman:
-            self.config.resman = 'host'
+        self.config.resman_options = self.popper_cfg.resource_manager.options
+        self.config.engine_options = self.popper_cfg.engine.options
```

Comment:
Lines 27-51 can be handled by the `PopperConfig` class. The idea is to hide all config-related functionality in that object, not only reading/parsing the configuration file. We can think of PopperConfig as an encapsulation of anything that has to do with configuration. The resulting object is passed around to access that information

Reply:
In other words. objects making use of the config object don't need to be aware that it's a `DotMap`. They only use the `config.foo` in order to access the value of `foo`

Reply:
I see. Ok.

Reply:
another way of looking at this. The idea would be to do:

```python
kwargs = locals()
kwargs.pop('self')
self.config = PopperConfig(**kwargs)
```

following code uses `self.config` directly without knowing it's backed by a `DotMap`. That means we can remove `dotmap` from being imported here.

Another thing that this might mean is that we can mover the `PopperConfig` from parser and put it in `popper.config`. This object represents the configuration, not only parses it.

Reply:
> As a folllowup of the above, we can move the PopperConfig from parser and put it in popper.config. This object represents the configuration, not only parses it.

You mean having a new `config.py` module right ? Yeah, thats makes sense.